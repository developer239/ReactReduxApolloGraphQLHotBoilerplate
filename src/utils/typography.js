import React from 'react'


export const nl2br = str => {
  const newlineRegex = /(\r\n|\r|\n)/g

  if (typeof str === 'number') {
    return str
  } else if (typeof str !== 'string') {
    return ''
  }

  return str.split(newlineRegex).map(function (line, index) {
    if (line.match(newlineRegex)) {
      // eslint-disable-next-line react/no-array-index-key
      return React.createElement('br', { key: index })
    }
    return line
  })
}
