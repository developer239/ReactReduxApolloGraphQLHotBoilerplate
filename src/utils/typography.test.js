import React from 'react'
import { mount } from 'enzyme'
import Provider from 'AppProvider'
import { nl2br } from './typography'


describe('Typography', () => {
  describe('nl2br', () => {
    it('renders without any new lines', () => {
      const wrapper = mount(
        <Provider>
          <div>{`Some simple text.`}</div>
        </Provider>
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('renders with new lines replaced by br elements', () => {
      const wrapper = mount(
        <Provider>
          <div>{nl2br(`Some simple

text.`)}</div>
        </Provider>
      )
      expect(wrapper).toMatchSnapshot()
    })
  })
})
