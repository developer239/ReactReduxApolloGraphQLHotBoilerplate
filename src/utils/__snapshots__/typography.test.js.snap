// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography nl2br renders with new lines replaced by br elements 1`] = `
<Provider
  mocks={Array []}
>
  <MockedProvider
    addTypename={false}
    mocks={Array []}
  >
    <ApolloProvider
      client={
        ApolloClient {
          "__operations_cache__": Map {},
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "clearStoreCallbacks": Array [],
          "clientAwareness": Object {},
          "defaultOptions": Object {},
          "disableNetworkFetches": false,
          "link": ApolloLink {
            "request": [Function],
          },
          "localState": LocalState {
            "cache": InMemoryCache {
              "addTypename": false,
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "config": Object {
                "addTypename": false,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "cacheKeyRoot": CacheKeyNode {
                  "children": null,
                  "key": null,
                },
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
            "client": [Circular],
          },
          "mutate": [Function],
          "query": [Function],
          "queryDeduplication": true,
          "reFetchObservableQueries": [Function],
          "resetStore": [Function],
          "resetStoreCallbacks": Array [],
          "ssrMode": false,
          "store": DataStore {
            "cache": InMemoryCache {
              "addTypename": false,
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "config": Object {
                "addTypename": false,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "cacheKeyRoot": CacheKeyNode {
                  "children": null,
                  "key": null,
                },
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
          },
          "typeDefs": undefined,
          "version": "2.5.1",
          "watchQuery": [Function],
        }
      }
    >
      <div>
        Some simple
        <br
          key="1"
        />
        <br
          key="3"
        />
        text.
      </div>
    </ApolloProvider>
  </MockedProvider>
</Provider>
`;

exports[`Typography nl2br renders without any new lines 1`] = `
<Provider
  mocks={Array []}
>
  <MockedProvider
    addTypename={false}
    mocks={Array []}
  >
    <ApolloProvider
      client={
        ApolloClient {
          "__operations_cache__": Map {},
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "clearStoreCallbacks": Array [],
          "clientAwareness": Object {},
          "defaultOptions": Object {},
          "disableNetworkFetches": false,
          "link": ApolloLink {
            "request": [Function],
          },
          "localState": LocalState {
            "cache": InMemoryCache {
              "addTypename": false,
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "config": Object {
                "addTypename": false,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "cacheKeyRoot": CacheKeyNode {
                  "children": null,
                  "key": null,
                },
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
            "client": [Circular],
          },
          "mutate": [Function],
          "query": [Function],
          "queryDeduplication": true,
          "reFetchObservableQueries": [Function],
          "resetStore": [Function],
          "resetStoreCallbacks": Array [],
          "ssrMode": false,
          "store": DataStore {
            "cache": InMemoryCache {
              "addTypename": false,
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "config": Object {
                "addTypename": false,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "cacheKeyRoot": CacheKeyNode {
                  "children": null,
                  "key": null,
                },
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
          },
          "typeDefs": undefined,
          "version": "2.5.1",
          "watchQuery": [Function],
        }
      }
    >
      <div>
        Some simple text.
      </div>
    </ApolloProvider>
  </MockedProvider>
</Provider>
`;
